toolchest_client/tools/tool.py

```diff
@@ -322,25 +339,49 @@ def _check_thread_health(self):
                 self._kill_query_threads()
                 raise ToolchestException("A job irrecoverably failed. See logs above for details.")
 
-    def _wait_for_threads_to_finish(self, check_health=True, wait_for_upload=True):
-        """Waits for all jobs and their corresponding threads to finish while printing their statuses."""
-        elapsed_seconds = 0
+    def _start_or_check_stream(self):
+        # Start streaming (in place of regular status updates).
+        if not self.streaming_asyncio_task:
+            # Clear the last status line before streaming begins
+            print("Printed lines:".ljust(120))
+            self.streaming_asyncio_task = asyncio.create_task(self.streaming_client.receive_stream())
+        # Check if an exception was raised in the streaming task
+        elif self.streaming_asyncio_task.done() and self.streaming_asyncio_task.exception():
+            self._kill_query_threads()
+            error_message = (
+                "A streaming error was encountered. See logs above for details.\n"
+                "Try re-running the same Toolchest command with this argument:\n"
+                "\tstreaming_enabled=False"
```

Comment:
Does this only happen with streaming errors, or on other worker node errors?

Reply:
This can happen with other worker node errors, but only if `streaming_enabled=True`.